test/core_tests.js

```diff
@@ -0,0 +1,138 @@
+QUnit.begin(function( details ) {
+  // In case tests didn't complete, we are left with a hash/anchor pointing somewhere. But we want to start from scratch:
+  window.location.hash = "";    
+});
+
+QUnit.test( "Initialize Impress.js", function( assert ) {
+  assert.ok( impress, 
+             "impress declared in global scope" );
+  assert.strictEqual( impress().init(), undefined,
+                    "impress().init() called." );
+  assert.strictEqual( impress().init(), undefined,
+                    "It's ok to call impress().init() a second time, it's a no-op." );
+  var notSupportedClass = document.body.classList.contains("impress-not-supported");
+  var yesSupportedClass = document.body.classList.contains("impress-supported");
+  if ( !_impressSupported() ) {
+    assert.ok( notSupportedClass,
+               "body.impress-not-supported class still there." );
+    assert.ok( !yesSupportedClass,
+               "body.impress-supported class was NOT added." );
+  } else {
+    assert.ok( !notSupportedClass,
+               "body.impress-not-supported class was removed." );
+    assert.ok( yesSupportedClass,
+               "body.impress-supported class was added." );
+               
+    // To be pedantic, we run the rest of these tests inside the else 
+    // brackets as well, meaning that impress.js is tested to be supported.
+    // However, other QUnit.test() blocks than this will fail miserably if it
+    // weren't supported.
+    assert.ok( !document.body.classList.contains("impress-disabled"),
+               "body.impress-disabled is removed." );
+    assert.ok( document.body.classList.contains("impress-enabled"),
+               "body.impress-enabled is added." );
+    
+    var canvas = document.querySelector( "div#impress > div" );
+    assert.ok( !canvas.classList.contains("step") && canvas.id === "",
+               "Additional 'canvas' div inserted between div#impress root and steps." );
+    assert.equal( canvas.style.transform,
```

Comment:
Also, it is a good practice to use strict equality when comparing two data types to prevents traps like this one:

```javascript
assert.equal( null, undefined ); // Passes, data is equal?
```

The solution to prevent this trap in the future is to use `strictEqual` instead:

```javascript
assert.strictEqual( null, undefined );// fails, data is different!
```